<a href="https://colab.research.google.com/github/Cubito918/data-analytics-portfolio/blob/main/clientes-analysis/clientes_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd

url = "https://raw.githubusercontent.com/Cubito918/data-analytics-portfolio/refs/heads/main/clientes-analysis/clientes.csv"

df = pd.read_csv(url)
df.head()


#cliente que más gastó
cliente_top = df.sort_values(by="total_gastado", ascending=False).iloc[0]
print(cliente_top)

#promedio de gasto
promedio = df["total_gastado"].mean()
print(promedio)

#método para segmentar clientes por nivel
def segmento(gasto):
    if gasto >= 10000:
        return "Alto valor"
    elif gasto >= 5000:
        return "Valor medio"
    else:
        return "Bajo valor"

df["segmento"] = df["total_gastado"].apply(segmento)
df

#conteo por segmento
df["segmento"].value_counts()

id_cliente              105
total_gastado         15000
frecuencia_compras        6
Name: 4, dtype: int64
7125.0


segmento
Valor medio    3
Bajo valor     3
Alto valor     2
Name: count, dtype: int64

##  Insights

- El cliente 105 pertenece al segmento **Alto Valor**.
- La mayoría de clientes se concentra en el segmento **Valor Medio**.
- Se recomienda priorizar retención de clientes de alto valor.